### PR TITLE
[FLINK-34482][checkpoint] Rename checkpointing options

### PR DESCRIPTION
**Pull request by masteryhx**
## What is the purpose of the change

Rename checkpoint related options

## Brief change log

  - Rename checkpoint related options with prefix 'execution.checkpointing'

## Verifying this change

This change is a trivial rework already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**yes** / no)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java [hunk 1]
```diff
@@ -97,11 +97,11 @@ public class CheckpointingOptions {
      *     to {@link #CHECKPOINTS_DIRECTORY}, and the checkpoint meta data and actual program state
      *     will both be persisted to the path.
      */
-    @Documentation.Section(value = Documentation.Sections.COMMON_STATE_BACKENDS, position = 2)
     public static final ConfigOption<String> CHECKPOINT_STORAGE =
-            ConfigOptions.key("state.checkpoint-storage")
```
Comment (Zakelly) [inline flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java:102]: Some `state.checkpoint-storage` in documents should also be updated.

**Review by Zakelly — CHANGES_REQUESTED** [review]
Thanks for the PR! I think the doc should also be re-generated.

File: flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java [hunk 2]
```diff
@@ -210,11 +209,11 @@ public class CheckpointingOptions {
      * <p>Local recovery currently only covers keyed state backends. Currently, MemoryStateBackend
      * does not support local recovery and ignore this option.
      */
-    @Documentation.Section(Documentation.Sections.COMMON_STATE_BACKENDS)
-    public static final ConfigOption<String> LOCAL_RECOVERY_TASK_MANAGER_STATE_ROOT_DIRS =
```
[the remark this answers is not in the document]
Comment (masteryhx) [reply flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java:214]: I just remain it as before.

File: flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java [hunk 3]
```diff
@@ -97,11 +97,11 @@ public class CheckpointingOptions {
      *     to {@link #CHECKPOINTS_DIRECTORY}, and the checkpoint meta data and actual program state
      *     will both be persisted to the path.
      */
-    @Documentation.Section(value = Documentation.Sections.COMMON_STATE_BACKENDS, position = 2)
```
[the remark this answers is not in the document]
Comment (masteryhx) [reply flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java:100]: I have introduced a new section `common_checkpointing` for them.

**Review by Zakelly — APPROVED** [review]
Overall LGTM. There still be some deprecated options used everywhere in docs. These are remaining problem and we can fix this later.

**Pull request closed without merging**